include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -217,10 +218,22 @@ class argument_parser
                     option_spec const & spec = option_spec::DEFAULT,
                     validator_type validator = validator_type{}) // copy to bind rvalues
     {
+        auto constexpr allowed = is_alnum || is_char<'_'> || is_char<'@'>;
         if (id_exists(short_id))
             throw parser_design_error("Option Identifier '" + std::string(1, short_id) + "' was already used before.");
         if (id_exists(long_id))
             throw parser_design_error("Option Identifier '" + long_id + "' was already used before.");
+        if (long_id.length() == 1)
+            throw parser_design_error("Long IDs must be either empty, or longer than one character.");
+        if (!allowed(short_id))
+            throw parser_design_error("Option identifiers may only contain alphanumeric characters, '_', or '@'.");
+        if (is_char<'-'>(long_id[0]))
```

Comment:
```suggestion
        if (long_id.size() > 0 && is_char<'-'>(long_id[0]))
```
otherwise this will segfault